sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala

```diff
@@ -168,6 +168,22 @@ class RocksDBSuite extends SparkFunSuite {
     }
   }
 
+  test("RocksDBFileManager: create init dfs directory with unknown number of keys") {
+    val dfsRootDir = new File(Utils.createTempDir().getAbsolutePath + "/state/1/1")
+    try {
+      val fileManager = new RocksDBFileManager(
+        dfsRootDir.getAbsolutePath, Utils.createTempDir(), new Configuration)
+      // Save a version of empty checkpoint files
+      val cpFiles = Seq()
+      assert(!dfsRootDir.exists())
+      saveCheckpointFiles(fileManager, cpFiles, version = 1, numKeys = -1)
```

Comment:
Could we verify the result with `loadAndVerifyCheckpointFiles(fileManager, verificationDir, version = 1, Nil, -1)`?

Reply:
Why we need to load the checkpoint here? The failure happens during snapshot.

Reply:
In majority of cases we do symmetric test - if you save checkpoint for verification, you are encouraged to load the checkpoint just you saved to verify it completely.

Reply:
Previously, I prefer pure unit test. And I see your points, already updated in the force pushed branch.